addons/mail/models/mail_message.py

```diff
@@ -346,7 +346,7 @@ def _message_read_dict_postprocess(self, messages, message_tree):
         }) for attachment in attachments_data)
 
         # 3. Tracking values
-        tracking_values = self.env['mail.tracking.value'].sudo().search([('mail_message_id', 'in', message_ids)])
+        tracking_values = self.env['mail.tracking.value'].sudo().search(['&', ('mail_message_id', 'in', message_ids), '|', ('company_id', '=', False), ('company_id', 'in', self.env.companies.ids)])
```

Comment:
`company_id', 'in', self.env.companies.ids`
I'm not sure it is correct, since the value displayed in view form will be based on current company (right?). The user can access information but according to the user active company, I think that the tracking message should reflect current record state.

Reply:
when two companies enable then in that case updated cost is enabled to that company who has updated the cost not to others this is the current scenario.
or 
updated cost changes according to the user company. 
What did you suggest?

Reply:
A company dependant field is computed based on env.company. Tracking should be consistent with that
Example use case:
User has two active companies: A & B 
Update company dependent field using company A From value 1 to value 2
Swith to company B
displayed field value will be 1 since the fields is update in company A not be. But since we are keeping all tracking values from actives companies (env.companies.ids)(A and B), the tracking values 1->2 will be displayed. Right? I didn't test that but this is confusing I think.

Reply:
I just noticed it was outdated, so it's all right